src/haz3lcore/statics/Mode.re

```diff
@@ -132,14 +132,16 @@ let ctr_ana_typ = (ctx: Ctx.t, mode: t, ctr: Constructor.t): option(Typ.t) => {
      a sum type having that ctr as a variant, we consider the
      ctr's type to be determined by the sum type */
   switch (mode) {
+  | Ana({term: Arrow(_, ty_out), _} as ty) =>
```

Comment:
is it intentional that this no longer normalizes?

Reply:
it didn't normalize before either. The situations I would expect to work even when the type annotation is not syntactically an arrow are still working for me. I'd like to eventually distinguish normalized and non-normalized types, it's quite complex to suss out what the invariants are.

Reply:
it didn't normalize before either. The situations I would expect to work even when the type annotation is not syntactically an arrow are still working for me. I'd like to eventually distinguish normalized and non-normalized types, it's quite complex to suss out what the invariants are.